spec/app/domain/authentication/authn-oidc/update_input_with_username_from_id_token_spec.rb

```diff
@@ -247,6 +268,141 @@ def request_body(request)
           expect { subject }.to raise_error(::Errors::Authentication::RequestBody::MissingRequestParam)
         end
       end
+
+      context "with valid id token in header" do
+        let(:audit_success) { false }
+
+        subject do
+          input_ = Authentication::AuthenticatorInput.new(
+            authenticator_name: 'authn-oidc',
+            service_id: 'my-service',
+            account: 'my-acct',
+            username: nil,
+            credentials: request_body(authenticate_id_token_request_id_token_in_header_field),
+            client_ip: '127.0.0.1',
+            request: authenticate_id_token_request_id_token_in_header_field
+          )
+
+          ::Authentication::AuthnOidc::UpdateInputWithUsernameFromIdToken.new(
+            verify_and_decode_token: mocked_decode_and_verify_id_token,
```

Comment:
The `mocked_decode_and_verify_id_token` mock accepts and parses JSON: https://github.com/cyberark/conjur/blob/master/spec/app/domain/authentication/authn-oidc/update_input_with_username_from_id_token_spec.rb#L18. 

It looks like this will fail if we pass Base64 encoded JSON. When the bearer token is passed in the body, is it passed as JSON?

Reply:
yes, we implemented same test as the `id_token` in the body.

Reply:
yes, we implemented same test as the `id_token` in the body.